### PR TITLE
Adding support for writing out reads for only a subset of indices in …

### PR DESCRIPTION
**Pull request by ktibbett**
…a lane.

### CODE REVIEW

**Comment by bradtaylor** [comment]
I looked this over, since I read through the code yesterday. Thumbs up from me. Jay can still review if you'd prefer his thumbsup.

**Comment by ktibbett** [comment]
Jay also saw this offline, so I'll call it good.  Thanks!

File: src/java/picard/illumina/IlluminaBasecallsConverter.java [hunk 1]
```diff
@@ -155,6 +157,8 @@ public int compare(final Integer integer1, final Integer integer2) {
 	 * @param codecPrototype         For spilling output records to disk.
 	 * @param outputRecordClass      Inconveniently needed to create SortingCollections.
 	 * @param includeNonPfReads      If true, will include ALL reads (including those which do not have PF set)
+     * @param ignoreUnexpectedBarcodes  If true, will ignore reads whose called barcode is not found in barcodeRecordWriterMap,
+     *                                  otherwise will throw an exception
```
Comment (nh13) [inline src/java/picard/illumina/IlluminaBasecallsConverter.java]: formatting

File: src/java/picard/illumina/IlluminaBasecallsToFastq.java [hunk 2]
```diff
@@ -153,6 +153,10 @@
     @Option(doc="Whether to include non-PF reads", shortName="NONPF", optional=true)
     public boolean INCLUDE_NON_PF_READS = true;
 
+    @Option(doc="Whether to ignore reads whose barcodes are not found in MULTIPLEX_PARAMS.  Useful when outputting " +
+            "fastqs for only a subset of the barcodes in a lane.", shortName="INGORE_UNEXPECTED")
```
Comment (nh13) [inline src/java/picard/illumina/IlluminaBasecallsToFastq.java]: optional?

File: src/java/picard/illumina/IlluminaBasecallsToSam.java [hunk 3]
```diff
@@ -201,6 +201,10 @@
     @Option(doc="Whether to include non-PF reads", shortName="NONPF", optional=true)
     public boolean INCLUDE_NON_PF_READS = true;
 
+    @Option(doc="Whether to ignore reads whose barcodes are not found in LIBRARY_PARAMS.  Useful when outputting " +
+            "BAMs for only a subset of the barcodes in a lane.", shortName="INGORE_UNEXPECTED")
```
Comment (nh13) [inline src/java/picard/illumina/IlluminaBasecallsToSam.java]: optional?

File: src/java/picard/illumina/IlluminaBasecallsToFastq.java (hunk 2 above)
Comment (ktibbett) [inline src/java/picard/illumina/IlluminaBasecallsToFastq.java]: Has a default value of false....

File: src/java/picard/illumina/IlluminaBasecallsToSam.java (hunk 3 above)
Comment (ktibbett) [inline src/java/picard/illumina/IlluminaBasecallsToSam.java]: Default value (that maintains current behavior) specified.

**Pull request merged**